projects/graphql-client/src/graphql-request.service.test.ts

```diff
@@ -90,159 +154,143 @@ describe('GraphQl Request Service', () => {
-    spectator.service.queryImmediately(modifiedRequest).subscribe();
-    // Return a response with same id but different value
-    const modifiedResponse = buildServerResponse();
-    modifiedResponse.data.test.value = 'baz';
-    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(modifiedResponse);
-    // Now make the first request again - it should be the original result
-    spectator.service.queryImmediately(buildRequest()).subscribe(result => {
-      expect((result as { value: string }).value).toEqual('bar');
+  test('merges requests together when possible', fakeAsync(() => {
+    const queryFnMock = mockQueryMethod();
+    queryFnMock.mockClear();
+    spectator.service.registerHandler(buildQueryHandlerOne());
+    spectator.service.query(buildRequestOne()).subscribe();
+    spectator.service.query(buildRequestOne('value2')).subscribe();
+    tick(debounceTimeMs);
+    expect(queryFnMock).toHaveBeenCalledTimes(1);
+    expect(queryFnMock).toHaveBeenCalledWith({
+      query: gql(
+        new GraphQlRequestBuilder()
+          .withSelects({ path: 'testone', children: [{ path: 'id' }, { path: 'value' }, { path: 'value2' }] })
+          .build()
+      ),
+      errorPolicy: 'all',
+      fetchPolicy: 'cache-first'
     });
-    // Tick, our responses are always async
-    tick();
   }));
 
-  test('only fires queries that are subscribed', () => {
-    spectator.service.registerHandler(buildQueryHandler());
-    spectator.service.queryImmediately(buildRequest());
-    spectator.controller.expectNone(graphQlUri);
-    spectator.service.queryImmediately(buildRequest()).subscribe();
-    spectator.expectOne(graphQlUri, HttpMethod.POST);
-  });
-
-  test('supports custom request options from query request handlers', () => {
-    const handler = buildQueryHandler({ cacheability: GraphQlRequestCacheability.NotCacheable });
+  test('supports custom request options from query request handlers', fakeAsync(() => {
+    const handler = buildQueryHandlerOne({ cacheability: GraphQlRequestCacheability.NotCacheable });
     spectator.service.registerHandler(handler);
-    spectator.service.queryImmediately(buildRequest()).subscribe();
-    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
-    // Make second identical request - should not be cached
-    spectator.service.queryImmediately(buildRequest()).subscribe();
-    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
-    expect(handler.getRequestOptions).toHaveBeenCalledTimes(2);
-  });
+    spectator.service.query(buildRequestOne()).subscribe();
+    const queryFnMock = mockQueryMethod();
+    tick(debounceTimeMs);
+    expect(queryFnMock).toHaveBeenCalledWith({
+      query: gql(
+        new GraphQlRequestBuilder()
+          .withSelects({ path: 'testone', children: [{ path: 'id' }, { path: 'value' }] })
+          .build()
+      ),
+      errorPolicy: 'all',
+      fetchPolicy: 'no-cache'
```

Comment:
In general, we want to assert what we're testing for in this specific test. the rest is just noise that needs to all be updated on each change. Here for example, we're testing that the request option is being propagated so we could just say
```typescript
    expect(queryFnMock).toHaveBeenCalledWith(expect.objectContaining({fetchPolicy: 'no-cache'}));`
```

Reply:
Yeah. But i wanted to test if errorpolicy is also present.